test/core.jl

```diff
@@ -5852,9 +5852,9 @@ let
     b3 = B23367[b][1] # copy b via array assignment
     addr(@nospecialize x) = ccall(:jl_value_ptr, Ptr{Cvoid}, (Any,), x)
     @test addr(b)  == addr(b)
-    @test addr(b)  == addr(b2)
-    @test addr(b)  == addr(b3)
-    @test addr(b2) == addr(b3)
+    # @test addr(b)  == addr(b2)
```

Comment:
@vtjnash, these tests are failing on this branch; is that expected? I'm not sure why the Ref `b2` case would be affected by anything I've changed, but I think I understand that for `b3`, because it's an array, when we store it in an array (inline now), then extract it back again, it's technically a different `jl_value_ptr`, but still the same bits.

Reply:
Ok, so it looks like in the Ref case, the alignment of `B23367` is now different, which is used by `datatype_pointerfree`, to determine how to `unsafe_convert` to a Ptr. Again, it seems like we're more correct now by saying that `B23367` is indeed `datatype_pointerfree`. I'm not sure how much people really run into this kind of thing because doing ccall w/ structs w/ Union fields probably isn't very common.